code/obj/item/device/pda2/ringtone.dm

```diff
@@ -0,0 +1,277 @@
+/// Ringtone Programs
+/datum/computer/file/pda_program/ringtone
+	name = "Ringtone"
+	size = 4
+	/// The list of ringtone datii included in this program. Assoc'd list, "ringtone_ID" = new/datum/ringtone/urtone
+	var/list/ring_list = list("ring1" = new/datum/ringtone)
+	/// The text that goes at the top
+	var/topText = "<h4>Thinktronic Systems PDA Sound System Backup</h4>"
+	/// The text that goes just under the header
+	var/bottomText = "In case you miss the classic."
+	/// The text that goes above the list of things
+	var/selectText = "Press \"Apply\" to restore the default PDA sound."
+	/// The thing that separates each of the entries on the list
+	var/dividerThing = "<hr>"
+	/// Include an option to override the PDA's alert message?
+	var/overrideAlertAllowed = 1
+	/// Should this ringtone's own alert message overwrite the PDA's?
+	var/overrideAlertMessage = 0
+	/// The text that asks the user whether to override the alert messages or not
+	var/overrideAlertText = "Use default ringer message?"
+	/// The text that indicates an affirmative for overriding the alert text
+	var/overrideAlertYesText = "Yes"
+	/// The text that indicates a negative for overriding the alert text
+	var/overrideAlertNoText = "No"
```

Comment:
Most of the funny overrides of the buttons are fine but imo this one's not really intuitive in what it does even when explained clearly and obscuring it with the overrides makes it reall confusing. I think I'd want this to not be overriden.

Reply:
I rewrote some of the prompts to be more descriptive with what it does. I still need it to be overridable, as not every alert message override is itself a typical PDA alert message.